File: langgraph/data/equiv/105.py
def arrange_by_number_name(numbers):
    number_names = {
        1: "One",
        2: "Two",
        3: "Three",
        4: "Four",
        5: "Five",
        6: "Six",
        7: "Seven",
        8: "Eight",
        9: "Nine",
    }
    descending_numbers = sorted(numbers, key=lambda x: -x)
    named_numbers = []
    index = 0
    while index < len(descending_numbers):
        num = descending_numbers[index]
        if num in number_names:
            named_numbers.append(number_names[num])
        index += 1
    return named_numbers

# method name: by_length -> arrange_by_number_name
# parameter name: arr -> numbers
# variable name: dic -> number_names, sorted_arr -> descending_numbers, new_arr -> named_numbers
# replaced for loop with while loop
# used lambda function for sorting in reverse order
# added explicit check for num in number_names dictionary

File: langgraph/result/qwen/7b-eq2js/105.js
function arrangeByNumberName(arr) {
    const numberNames = {
        1: "One",
        2: "Two",
        3: "Three",
        4: "Four",
        5: "Five",
        6: "Six",
        7: "Seven",
        8: "Eight",
        9: "Nine"
    };
    const descendingNumbers = arr.sort((a, b) => b - a);
    const namedNumbers = [];
    let index = 0;
    while (index < descendingNumbers.length) {
        const num = descendingNumbers[index];
        if (numberNames.hasOwnProperty(num)) {
            namedNumbers.push(numberNames[num]);
        }
        index++;
    }
    return namedNumbers;
}

// method name: by_length -> arrangeByNumberName
// parameter name: arr -> numbers
// variable name: dic -> numberNames, sorted_arr -> descendingNumbers, new_arr -> namedNumbers
// replaced for loop with while loop
// used sort function with comparator for sorting in reverse order
// added explicit check for num in numberNames object using hasOwnProperty method